clippy_lints/src/matches.rs

```diff
@@ -2266,7 +2276,18 @@ fn lint_match_arms<'tcx>(cx: &LateContext<'tcx>, expr: &Expr<'_>) {
                             ),
                         );
                     } else {
-                        diag.span_help(i.pat.span, &format!("consider refactoring into `{} | {}`", lhs, rhs));
+                        // Alignment is required to put the second sentence onto the next line,
+                        // keeping the indentation of the first sentence (with respect to the `help: ` prefix).
+                        let alignment = "\n      ";
+                        let message = format!(
+                            "consider refactoring into `{lhs} | {rhs}`{align}\
+                            If arms are supposed to be the same, consider making them visually distinguishable",
```

Comment:
You can just add another `diag.help("...or consider changing the match arm bodies")`.